StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

```diff
@@ -1527,11 +1527,21 @@ public class STPPaymentHandler: NSObject {
                         timeout: pollingBudget?.networkTimeout
                     ) { [self] paymentIntent, error in
                         guard let paymentIntent, error == nil else {
-                            let error = error ?? self._error(for: .unexpectedErrorCode, loggingSafeErrorMessage: "Missing PaymentIntent.")
-                            currentAction.complete(
-                                with: STPPaymentHandlerActionStatus.failed,
-                                error: error as NSError
-                            )
+                            // If we got an error retrieving the intent, retry if budget allows
+                            if let pollingBudget, pollingBudget.canPoll {
```

Comment:
This will only retry _if_ we have a budget, which means it's a polling call, we won't retry if it's the first call. This is consistent w/ master retry logic, but we could do like `pollingBudget ?? PollingBudget(duration: someArbitraryDuration)` to retry first calls to.

Reply:
I'm okay to call it good enough here but maybe we can add some more comments explaining, I'm hoping we can rewrite PaymentHandler next year and I don't want to lose track of this.

"This will only retry if we have a budget, which means it's a polling call, we won't retry if it's the first call. Ideally we retry on the first call too" or something